pyccl/tests/test_cclobject.py

```diff
@@ -4,6 +4,25 @@
 import functools
 
 
+def test_fancy_repr():
+    # Test fancy-repr controls.
+    cosmo1 = ccl.CosmologyVanillaLCDM()
+    cosmo2 = ccl.CosmologyVanillaLCDM()
+
+    ccl.CCLObject._fancy_repr.disable()
+    assert "False" in repr(ccl.CCLObject._fancy_repr)
+    assert repr(cosmo1) == object.__repr__(cosmo1)
+    assert cosmo1 != cosmo2
+
+    ccl.CCLObject._fancy_repr.enable()
+    assert "True" in repr(ccl.CCLObject._fancy_repr)
+    assert repr(cosmo1) != object.__repr__(cosmo2)
+    assert cosmo1 == cosmo2
+
+    with pytest.raises(AttributeError):
+        cosmo1._fancy_repr.disable()
```

Comment:
Would `ccl.Cosmology._fancy_repr.disable()` also fail? (I.e. calling from the class instead of the instance)

Reply:
Yes, every instance of `CCLObject` would fail. Instances might be subclasses, or instances of subclasses.

Reply:
I'll add a test for the class.

Reply:
Done, check and we can merge.